### ISSUE TITLE
Error code : 101 whitebox

### ISSUE THREAD

**Issue by mathiaschabal** [jblindsay/whitebox-tools#299]
Hello,

I am currently trying to use the withebox LIDAR tools on my mac via the Gui (same error with Qgis), but i have a consistant error code, the 101 code.

**What I Did**

I am using Lidar data from the French national geographic institut, the file is under the .laz extension. I tried three different tools in withebox (lidarinfo, ClipLidarToPolygon, ClassifyBuldingLidar) and the same error code appears. The wired thing is that i downloaded a different set of lidar data from the same platform i downloaded the previous set and the error did no occur on that set even though i followed the same protocol. The two data sets are at two different geographic locations (300 km a part).

I have attached the screenshot of the parameters and the error that occurs.

![Capture d’écran 2022-11-28 à 11 52 00](https://user-images.githubusercontent.com/119277226/204336155-2e8d61fa-ef0a-4b37-8992-068f5b8507d6.png)
![Capture d’écran 2022-11-28 à 11 52 14](https://user-images.githubusercontent.com/119277226/204336157-e80f8959-a32d-4e95-890f-f7178ea5b03e.png)
![Capture d’écran 2022-11-28 à 11 52 18](https://user-images.githubusercontent.com/119277226/204336161-f5645bcd-d9da-4080-95aa-a9c0eaa1262c.png) 

Does anyone have an idea why this might happen ?

**Comment by jblindsay** [comment]
Are you able to share the file that caused the error with me so that I may investigate? The error looks to be related to a buffer error when decompressing the LAZ file and so I suspect it's an issue with the LAZ reader/writer, but I'll need to confirm.

**Comment by mathiaschabal** [comment]
I will like to but the file is too big to be loaded here on github, it says. I am new to github, is their any another way for me to transfer you the file ?

**Comment by jblindsay** [comment]
Given that it is a file buffer error and you're telling me that the file is very large, might it be a memory issue?

**Comment by jblindsay** [comment]
How big is the file?

**Comment by mathiaschabal** [comment]
big, to big for github. It's 315 mo

**Comment by mathiaschabal** [comment]
the over file that i tested is roughly the same size.

**Comment by jblindsay** [comment]
315MB isn't that big, although given it's a compressed LAZ file, it may well decompress to something very large indeed. How much memory do you have on your system? Do you know how many points the file contains? If you can share a link to the file, I can take a look. It looks like you have it saved on OneDrive. Are you able to share external to the institution on your OneDrive?

**Comment by mathiaschabal** [comment]
My mac has 8gb of memory, i tried on my university computer that has 32gb of memory and it does the same problem. 
To be able to share the file i need an email address

**Comment by mathiaschabal** [comment]
In fact no here is the link : https://inrs-my.sharepoint.com/:f:/g/personal/mathias_chabal_inrs_ca/EjAbZN6WqAJFu0KCo4i4OeAB28gXujDN5UNxWFERT_9VYw?e=Wl4k5i

**Comment by jblindsay** [comment]
There's your problem. That's not nearly enough RAM to be processing LiDAR data of that size.

**Comment by mathiaschabal** [comment]
Yes but i tried on an other computer that has 32gb and it didnt work, i tried with another lidar file that is the same size and it dosent do that error.

**Comment by jblindsay** [comment]
The files are LAZ files, meaning that they are compressed. Often LAZ files are about 10% of the size of the decompressed data, so we're looking at potentially 3GB files here. Each tool, depending on the specifics of it, may require multiples of the decompressed size. Clearly you are pushing it close if not exceeding with the 8GB RAM machine. The fact that one similar sized file worked and another didn't likely means that the one file had a poorer compression ratio than the other and therefore was smaller when decompressed. Anyhow, I'm downloading the data that you sent now, but it looks like it'll take a while. Is there one specific file in the folder that caused you issue?

**Comment by mathiaschabal** [comment]
you can take any file, they all have the same problem

**Comment by mathiaschabal** [comment]
dont take care of the file starting with "ept" that is QGIS generating them

**Comment by jblindsay** [comment]
Okay, I can confirm that the buffer error occurs on my machine, with 64GB memory, as well. I'll look into it more when I get an opportunity. Thanks for reporting.

**Comment by jblindsay** [comment]
It looks like the issue was in the library that I use for reading/writing LAZ files. After I updated it from v0.7.4 to v0.7.8 it seems that it now is able to read each of the LAZ files that you have provided without issue. Thanks again for reporting this issue. The fix is now in the main branch and will be released with the next version of WhiteboxTools, which will likely be early in the new year (we're aiming for a January release). If you can't wait that long, you can compile from source code.
<img width="843" alt="Screenshot 2022-11-28 at 4 21 11 PM" src="https://user-images.githubusercontent.com/8441542/204383481-962e9b91-08bd-4a84-9d12-ef7e3fd8efd8.png">

**Comment by mathiaschabal** [comment]
Okay ! thank you very much for your very fast support, it's much appreciated !!

I can wait for the release, it is not urgent at all.
